### ISSUE TITLE
Wandb TRLX blogpost has incorrect losses for reward model

### ISSUE THREAD

**Issue by jihan-yin** [CarperAI/trlx#321]
### 🐛 Describe the bug

I'm currently using a variant of the TRLX code to train a reward model using DeBerta-Large. It's not going well as the loss is not going down, so to help debug, I took a look at the [blog post on wandb](https://wandb.ai/carperai/summarize_RLHF/reports/Implementing-RLHF-Learning-to-Summarize-with-trlX--VmlldzozMzAwODM2#reward-model) written by @PhungVanDuy and others. 

I noticed that the train loss plots go from 3 to 2.1. This doesn't make sense to me, as the loss is the per pair average of `-log(sigmoid(good_rewards - bad_rewards))`. First off, in the beginning, the rewards should be similar to each other, with an expected difference of 0 within the sigmoid. This leads to a sigmoid value of 0.5, which should lead to a loss of 0.7. A loss of 3 implies a sigmoid value of `e^-3 = 0.05`, which means the difference in rewards is very negative. If there's such a huge average difference in rewards, the model should already be good to go!

This logic extends to the loss stabilizing at 2.2. Plugging it into the loss function, it still doesn't make sense. 

Edit: The loss plots I'm describing are not from my training of deberta. It is from the blog post on wandb.

**Comment by PhungVanDuy** [comment]
> 

Thank you for your feedback.

I don't know what you mean by an incorrect loss here. Your trained DeBerta-Large (encoder model) is quite different compared with our setting for the reward model, I used GPT-J initialized from the SFT model to introduce the reward model. With pair average loss, I believe there have several variances to calculate the loss, actually, we did try experiments with several options to calculate loss, and the setting above got the best accuracy. In terms of the reward model, Alex also made great repo for the reward model [here](https://github.com/Dahoas/reward-modeling/blob/main/reward-modeling/finetune_rm.py#L16) there provided several options that you can try.

**Comment by jihan-yin** [comment]
Hey - the Deberta-Large stuff is kind of irrelevant to this issue, it's just context for why I started looking at the blog post. The loss I'm talking about are the loss plots shown in the blogpost, which doesn't seem to agree with the loss function defined in the code.

**Comment by jihan-yin** [comment]
> I believe there have several variances to calculate the loss
Does this mean the loss plots shown in the blog post were calculated using code that was different than what was shown in the blog post?

**Comment by PhungVanDuy** [comment]
> 

It uses the same loss, not a difference between train and validation, as you can check with the code on the repo. The wandb has problems (may train loss divided by 10), if you check the steps it does not align with our training steps (our training steps are around a thousand not a few hundred) like the chart. The loss of validation makes sense to me, loss of training should be in the same range. Thank you for great finding. We will update the new wandb.

**Comment by jihan-yin** [comment]
Gotcha, thanks for the quick response! Yeah the validation loss looks correct to me.
